spec/models/player_spec.rb

```diff
@@ -0,0 +1,693 @@
+        },
+        "3" => {
+          "2" => "lo (3b) max",
+          "3" => "gb (1b) C",
+          "4" => "SINGLE (cf)",
+          "5" => "popout (ss)",
+          "6" => "fly (lf) B?",
+          "7" => "popout (ss)",
+          "8" => "fly (cf) B?",
+          "9" => "lineout (3b)",
+          "10" => "popout (ss)",
+          "11" => "foulout (c)",
+          "12" => "HBP plus injury"
+        }
+      },
+      vs_rhp: {
+        "1" => {
+          "2" => "lo (3b) max",
+          "3" => "lineout (3b)",
+          "4" => "lineout (3b)",
+          "5" => "fly (lf) B?",
+          "6" => "SINGLE**",
+          "7" => "strikeout",
+          "8" => {
+            "1-7" => "SI*",
+            "8-20" => "lo (3b)"
+          },
+          "9" => "fly (lf) B?",
+          "10" => "popout (2b)",
+          "11" => "popout (2b)",
+          "12" => "HBP plus injury"
+        },
+        "2" => {
+          "2" => "fly (rf) B?",
+          "3" => "gb (1b) C",
+          "4" => {
+            "1-12" => "HR",
+            "13-20" => "TR"
+          },
+          "5" => {
+            "1-4" => "TR",
+            "5-20" => "DO"
+          },
+          "6" => "DOUBLE (lf)",
+          "7" => {
+            "1-7" => "DO**",
+            "8-20" => "SI**"
+          },
+          "8" => "WALK",
+          "9" => "gb (2b) C",
+          "10" => "WALK",
+          "11" => "SINGLE(lf)",
+          "12" => {
+            "1-16" => "HR",
+            "17-20" => "fly (lf) B"
+          }
+        },
+        "3" => {
+          "2" => "fly (lf) B?",
+          "3" => "foulout (c)",
+          "4" => "SINGLE**",
+          "5" => "popout (ss)",
+          "6" => "fly (cf) B?",
+          "7" => "gb (ss) B+",
+          "8" => "popout (ss)",
+          "9" => "popout (ss)",
+          "10" => "lineout (3b)",
+          "11" => "foulout (3b)",
+          "12" => "WALK"
+        }
+      }
+    )
+  end
+
+  it { should belong_to(:team) }
+
+  it { should validate_presence_of(:team) }
+  it { should validate_presence_of(:first_name) }
+  it { should validate_presence_of(:last_name) }
+  it { should validate_presence_of(:batting_hand) }
+  it { should validate_presence_of(:stealing) }
+  it { should validate_presence_of(:running) }
+  it { should validate_presence_of(:vs_lhp) }
+  it { should validate_presence_of(:vs_rhp) }
+  it { should validate_presence_of(:avg) }
+  it { should validate_presence_of(:ab) }
+  it { should validate_presence_of(:doubles) }
+  it { should validate_presence_of(:triples) }
+  it { should validate_presence_of(:hr) }
+  it { should validate_presence_of(:rbi) }
+  it { should validate_presence_of(:bb) }
+  it { should validate_presence_of(:so) }
+  it { should validate_presence_of(:sb) }
+  it { should validate_presence_of(:cs) }
+  it { should validate_presence_of(:slg) }
+  it { should validate_presence_of(:obp) }
+
+  describe "#vs_lhp" do
+    context "missing white die value" do
+      subject do
+        Player.new(
```

Comment:
There's gotta be a way to DRY these up a bit. The line count of this spec is going to be enormous as we beef this up if we continue to follow the existing pattern.